lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer');
+
+/**
+ * @fileoverview Tracks unused CSS rules.
+ */
+class JsUsage extends Gatherer {
+  beforePass(options) {
+    return options.driver.sendCommand('Profiler.enable')
+      .then(_ => options.driver.sendCommand('Profiler.startPreciseCoverage'))
+      .catch(err => {
+        // TODO(phulce): Remove this once JS usage hits stable
+        if (/startPreciseCoverage/.test(err.message)) {
+          throw new Error('Js Usage tracking requires Chrome \u2265 58');
```

Comment:
I might suggest Chrome 58+ for future-proofing. I'm also unsure whether (somewhere) we need to capture that coverage is function level vs line-level. It might matter less to folks.